neuralpp/symbolic/sympy_expression.py

```diff
@@ -113,7 +133,10 @@ def _sympy_function_to_python_callable(sympy_function: sympy.Basic) -> Callable:
         return sympy_function_python_callable_dict[sympy_function]
     except KeyError:
         if sympy_function == sympy.Piecewise:
-            return functions.conditional
+            return sympy_function
```

Comment:
when converting, keep `piecewise` and `conditional` distinct. So it's possible a `BasicFunctionApplication` has `sympy.Piecewise` as its `function`.